contracts/vault/packages/VaultPackage.sol

```diff
@@ -85,13 +85,25 @@ contract VaultPackage is VaultStorage, IVault, IVaultInit, IVaultEvents {
         if (length > MAX_QUEUE) {
             revert QueueTooLong();
         }
-        // Make sure every strategy in the new queue is active.
-        for (uint256 i = 0; i < newDefaultQueue.length; i++) {
+
+        // Make sure every strategy in the new queue is active and not duplicated.
+        for (uint256 i = 0; i < length; i++) {
             address strategy = newDefaultQueue[i];
+            
+            // Check for active strategy.
             if (strategies[strategy].activation == 0) {
                 revert InactiveStrategy(strategy);
             }
+            
+            // Check for duplicates by comparing with the rest of the queue.
+            // Introduces a O(n^2) complexity but the queue is expected to be small.
+            for (uint256 j = i + 1; j < length; j++) {
+                if (strategy == newDefaultQueue[j]) {
+                    revert DuplicateStrategy(strategy);
+                }
+            }
         }
+
```

Comment:
We need a proper test for all those cases.

Reply:
Included on last commit